cpp/examples/ide_secir_ageres.cpp

```diff
@@ -0,0 +1,136 @@
+    vec_init[num_transitions + (int)mio::isecir::InfectionTransition::InfectedCriticalToRecovered]          = 1.0;
+    vec_init = vec_init * dt;
+    // Add initial time point to time series.
+    init.add_time_point(-10, vec_init);
+    // Add further time points until time 0.
+    while (init.get_last_time() < -dt / 2) {
+        init.add_time_point(init.get_last_time() + dt, vec_init);
+    }
+
+    // Initialize model.
+    mio::isecir::Model model(std::move(init), N, deaths, num_agegroups, std::vector<ScalarType>(num_agegroups, 100.));
+
+    // Uncomment these lines to use a different method to initialize the model using the TimeSeries init.
+    // Initialization method with Susceptibles.
+    //model.m_populations.get_last_value()[(Eigen::Index)mio::isecir::InfectionState::Susceptible] = 1000;
+    // model.m_populations.get_last_value()[(Eigen::Index)mio::isecir::InfectionState::Count +
+    //                                      (Eigen::Index)mio::isecir::InfectionState::Susceptible] = 1000;
+    // Initialization mehtod with Recovered.
+    //model.m_populations.get_last_value()[(Eigen::Index)mio::isecir::InfectionState::Recovered] = 0;
+    // model.m_populations.get_last_value()[(Eigen::Index)mio::isecir::InfectionState::Count +
+    //                                      (Eigen::Index)mio::isecir::InfectionState::Recovered] = 0;
+
+    // Set working parameters.
+    // First AgeGroup for Transition Distributions.
+    mio::SmootherCosine smoothcos1(2.0);
+    mio::StateAgeFunctionWrapper delaydistribution1(smoothcos1);
+    std::vector<mio::StateAgeFunctionWrapper> vec_delaydistrib1(num_transitions, delaydistribution1);
+    // TransitionDistribution is not used for SusceptibleToExposed. Therefore, the parameter can be set to any value.
+    vec_delaydistrib1[(int)mio::isecir::InfectionTransition::SusceptibleToExposed].set_distribution_parameter(-1.);
+    vec_delaydistrib1[(int)mio::isecir::InfectionTransition::InfectedNoSymptomsToInfectedSymptoms]
+        .set_distribution_parameter(3.0);
+
+    model.parameters.get<mio::isecir::TransitionDistributions>()[mio::AgeGroup(0)] = vec_delaydistrib1;
+
+    //Second AgeGroup for Transition Distributions.
+    mio::SmootherCosine smoothcos2(3.0);
+    mio::StateAgeFunctionWrapper delaydistribution2(smoothcos2);
+    std::vector<mio::StateAgeFunctionWrapper> vec_delaydistrib2(num_transitions, delaydistribution2);
+    // TransitionDistribution is not used for SusceptibleToExposed. Therefore, the parameter can be set to any value.
+    vec_delaydistrib2[(int)mio::isecir::InfectionTransition::SusceptibleToExposed].set_distribution_parameter(-1.);
+    vec_delaydistrib2[(int)mio::isecir::InfectionTransition::InfectedNoSymptomsToInfectedSymptoms]
+        .set_distribution_parameter(4.0);
+
+    model.parameters.get<mio::isecir::TransitionDistributions>()[mio::AgeGroup(1)] = vec_delaydistrib2;
+
+    for (auto group = mio::AgeGroup(0); group < mio::AgeGroup(num_agegroups); ++group) {
```

Comment:
Throughout the code you are switching the type you are using whenever you are iterating over all age groups. Does the type here have to be AgeGroup or can you also use Eigen::Index or size_t? In other cases you are using a lot of type conversions and I was wondering if this is necessary or if you could use just another type in the for loop. We can also discuss this together :)